app/use_cases/update_child_person.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class UpdateChildPerson < ActiveInteraction::Base
+  object :person
+  string :email, default: nil
+  string :first_name
+  string :role
+
+  # Updates a child person
+  def execute
+    unless person.update(email: email,
+                         first_name: first_name,
+                         role: find_role)
+      errors.merge!(person.errors)
+    end
+    person
+  end
+
+  private
+
+  def find_role
+    team = find_team
+
+    role_obj = team.roles.find_by(title: role)
+    return rerr("Specified role '#{role}' not found in team #{team.name}") if role_obj.blank?
+
+    role_obj
+  end
+
+  def find_team
+    team = person.role.team
+    return rerr('Team not found') if team.blank?
+
+    team
+  end
+
+  def rerr(msg)
```

Comment:
didn't you have a general function for this?

Reply:
No that was the logg function.